pep-0692.rst

```diff
@@ -198,19 +211,52 @@ Continuing the previous example:
         number: NotRequired[int]
     
     def src(**kwargs: **Example): ...
-    def dest(animal: Dog, string: str, number: int = ...): ...
+    def dest(*, animal: Dog, string: str, number: int = ...): ...
 
     dest = src  # OK!
 
-3. The destination callable contains ``**kwargs: **TypedDict``, the source
-callable doesn't contain ``**kwargs`` and the destination function's
-``TypedDict`` fields are assignable to the corresponding keyword arguments in
-the source function. Again, required and not required keys should be compatible
-with the keyword arguments and the rest of the parameters should be compatible.
-For example:
+It is worth pointing out that the destination function's arguments that are to
+be compatible with the keys and values from the ``TypedDict`` must be keyword
+only arguments:
 
 ::
 
+    def dest(animal: Dog, string: str, number: int = ...): ...
+    dest(animal_instance, "some string")  # OK!
+    dest = src
+    dest(animal_instance, "some string")  # WRONG! The same call fails at
+                                          # runtime now because 'src' expects
+                                          # keyword arguments.
+
+3. The reverse situation where the destination callable
+contains ``**kwargs: **TypedDict`` and the source callable doesn't contain
+``**kwargs`` should be only allowed if the ``TypedDict`` in the destination
+function signature is final. This is because, we cannot be sure that additional
+keyword arguments are not being passed in when an instance of a subclass had
+been assigned to a variable with a base class type and then unpacked in the
+destination callable invocation:
+
+::
+
+    def dest(**Animal): ...
+    def src(name: str): ...
+
+    dog: Dog = {"name": "Daisy", "breed": "Labrador"}
+    animal: Animal = dog
+
+    dest = src      # WRONG!
+    dest(**animal)  # Fails at runtime.
+
+However, when the ``TypedDict`` in the function signature is final, we are
```

Comment:
Question: The idea that decorating a TD class with @final ensures its values will not have additional keys doesn’t make sense to me. PEP 589 says that TD compatibility is based on structural, not nominal, subtyping. So whether another TD *subclasses* the intended one (as prevented by final) is irrelevant. 

Here’s an [example](https://mypy-play.net/?mypy=latest&python=3.10&gist=084cc24abbae44aa982bd2255cb325d5).

Reply:
Right, I was thinking about that, thanks for mentioning that. At the time it seemed like a good idea because I thought that as a user you would have to "hack around" to break that intentionally, but probably that's not a good premise to base the specification on. On second thought it seems like it would make more sense to disallow that. Let me make the changes.